tools/AutoTuner/pyproject.toml

```diff
@@ -1,26 +1,37 @@
 [project]
 name = "autotuner"
 version = "0.0.1"
+requires-python = ">=3.8, <3.11"
 description = "This project provides a set of tools for tuning OpenROAD-flow-scripts parameter without user interference."
 classifiers = [
     "Programming Language :: Python :: 3",
     "License :: OSI Approved :: BSD 3-Clause",
 ]
 readme = "README.md"
-requires-python = ">= 3.8"
-dynamic = ["dependencies", "optional-dependencies"]
+dependencies = [
+    "ray[default,tune] >= 2.10, <= 2.23",
```

Comment:
To close #2859 we need at least `2.40`. We can continue with the PR, but this will not be enough to close the issue.